### PR TITLE
Added new TC DeleteBackupSharedByMultipleUsersFromAdmin

### PR DESCRIPTION
**Pull request by ak-px**
**What this PR does / why we need it**:
Added new DeleteBackupSharedByMultipleUsersFromAdmin and Required Helper Functions
DeleteBackupSharedByMultipleUsersFromAdmin - https://portworx.testrail.net/index.php?/cases/view/87565

**Which issue(s) this PR fixes** (optional)
Closes #PA-1407

**Special notes for your reviewer**:
This PR doesn't contains 2.5.2 api
logs:
s3:
https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/system-tests/job/basic-system-test/job/px/job/NFS/job/PX-Cloud-Vanilla-NFS/134/

Aetos Dashboard: https://aetos.pwx.purestorage.com/resultSet/testSetID/332021

NFS:
https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/system-tests/job/basic-system-test/job/px/job/NFS/job/PX-Cloud-Vanilla-NFS/136/

Aetos Dashboard: https://aetos.pwx.purestorage.com/resultSet/testSetID/332160

### CODE REVIEW

**Review by snigdha-px — APPROVED** [review]
Minor comment

**Comment by ak-px** [comment]
@snigdha-px , Thanks for reviewing, Addressed all your comments.

File: tests/backup/backup_orphaned_delete_test.go [hunk 1]
```diff
@@ -0,0 +1,362 @@
...
+
+		Step("Create a non-admin users to create the backups and restore", func() {
+			log.InfoD(fmt.Sprintf("Create a non-admin users to create the backups and restore"))
+			roles := [3]backup.PxBackupRole{backup.ApplicationOwner, backup.InfrastructureOwner, backup.ApplicationUser}
+			for _, role := range roles {
+				userName = fmt.Sprintf("testauser-%v-%s", role, RandomString(4))
+				firstName := fmt.Sprintf("FirstName-%v-%s", role, RandomString(4))
+				lastName := fmt.Sprintf("LastName-%v-%s", role, RandomString(4))
+				email := fmt.Sprintf("testuser-%v-%s@cnbu.com", role, RandomString(4))
+				err := backup.AddUser(userName, firstName, lastName, email, commonPassword)
```
Comment (kphalgun-px) [inline tests/backup/backup_orphaned_delete_test.go:85]: How about?

`userName = createUsers(1)[0]` just to be consistent?

File: tests/backup/backup_orphaned_delete_test.go [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+				log.FailOnError(err, "Failed to create user - %s", userName)
```
Comment (kphalgun-px) [inline tests/backup/backup_orphaned_delete_test.go:86]: How about?

userName = createUsers(1)[0] just to be consistent?

File: tests/backup/backup_orphaned_delete_test.go [hunk 3]
```diff
@@ -0,0 +1,362 @@
...
+			log.FailOnError(err, "Fetching px-admin ctx")
+			for _, provider := range providers {
+				adminCloudCredUID = uuid.New()
+				adminCredName = fmt.Sprintf("autogenerated-cred-%v", time.Now().Unix())
+				err = CreateCloudCredential(provider, adminCredName, adminCloudCredUID, orgID, adminCtx)
+				log.FailOnError(err, "Failed to create cloud credential - %s", err)
+				log.Infof("Updating cloud credential ownership for non-admin users")
+				if provider != drivers.ProviderNfs {
+					err = UpdateCloudCredentialOwnership(adminCredName, adminCloudCredUID, []string{appAdminUserName}, nil, Read, Invalid, adminCtx, orgID)
+					dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying updation of ownership for cloudcredential- %s", adminCredName))
```
Comment (kphalgun-px) [inline tests/backup/backup_orphaned_delete_test.go:115]: How about including username in the log message?

dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying ownership update for cloud credential %s to user %s", adminCredName, []string{appAdminUserName}))

File: tests/backup/backup_orphaned_delete_test.go [hunk 4]
```diff
@@ -0,0 +1,362 @@
...
+			adminCtx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-admin ctx")
+			for _, provider := range providers {
+				backupLocationName = fmt.Sprintf("autogenerated-backup-location-%v", time.Now().Unix())
+				backupLocationUID = uuid.New()
+				err := CreateBackupLocationWithContext(provider, backupLocationName, backupLocationUID, adminCredName, adminCloudCredUID, getGlobalBucketName(provider), orgID, "", "", adminCtx)
+				log.FailOnError(err, "Failed to add backup location %s using provider %s for px-admin user", backupLocationName, provider)
+				log.Infof("Updating backup location ownership for non-admin users")
+				err = UpdateBackupLocationOwnership(backupLocationName, backupLocationUID, []string{appUserName}, nil, Read, Invalid, adminCtx)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying updation of ownership for backupLocation - %s", backupLocationName))
```
Comment (kphalgun-px) [inline tests/backup/backup_orphaned_delete_test.go:131]: How about including username in the log message?

dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying ownership update for backup location %s to user %s", backupLocationName, []string{appUserName}))

**Review by kphalgun-px — APPROVED** [review]
LGTM

**Comment by ak-px** [comment]
@kphalgun-px , Addressed all your comments, Please re-review it, Thanks.

**Comment by ak-px** [comment]
Thanks for the reviews. These changes are now included in PR https://github.com/portworx/torpedo/pull/1669.

**Pull request closed without merging**